src/tokenizer.rs

```diff
@@ -406,7 +434,18 @@ impl<'a> Tokenizer<'a> {
                 '|' => {
                     chars.next(); // consume the '|'
                     match chars.peek() {
-                        Some('|') => self.consume_and_return(chars, Token::StringConcat),
+                        Some('/') if dialect_of!(self is PostgreSqlDialect) => {
```

Comment:
@nickolay, am I right that this is not going to work with custom dialect as you suggest here https://github.com/ballista-compute/sqlparser-rs/pull/243#discussion_r461094779?

Reply:
I have custom `Dialect` https://github.com/alex-dukhno/database/blob/master/src/parser/src/lib.rs#L221-L231, is it true that it wouldn't parse `|/` into `SquareRoot` because of `if dialect_of!(self is PostgreSqlDialect)`?

Reply:
Ah, I see now. Yes, I assumed `|/` is something very Postgres-specific, so limiting it to PostgreSqlDialect _(and GenericDialect, which I didn't mention explicitly)_ seemed to make sense.

In the older comment you're referring to I described a stop-gap measure that would be used until someone figured out the proper way to deal with `$...` in the parser.

We could make the new operators parse unconditionally for now if that makes your life easier, but without a clear use-case (and accompanying testcase) this will likely break in the future.

Reply:
I think it's better to have `dialect_of` checks and have a solution for dealing with `$...`. Having my knowledge of crate I'd suggest to add `Variable(Ident)` or something similar to `ast::Expr` enum. It has to be done in a separate PR of course. WDYT?

Reply:
I have @-mentioned you in #265, where this discussion can continue.

The hard part is describing what specifically we're trying to implement and why. Your suggestion, for example, solves the problem of representing `$` (but not another leader) followed by an identifier _(quoted or not; consisting of characters from an unspecified set)_ anywhere an arbitrary expression is allowed _(but not in more restricted contexts, like `OFFSET __ ROWS`)._ I'm not sure if this matches Postgres or the "common denominator" of the usual dialects; answering that requires some research. From the examples I've seen so far, a new `Value` variant with `{leader, ident}` fields might be more appropriate.